src/event-mixin.js

```diff
@@ -196,14 +196,32 @@ function retargetNonBubblingEvent(e) {
   });
   for (let i = path.length - 1; i >= 0; i--) {
     node = path[i];
+    // capture phase fires all capture handlers
     fireHandlers(e, node, 'capture');
     if (e.__propagationStopped) {
       return;
     }
   }
-  Object.defineProperty(e, 'eventPhase', {value: Event.BUBBLING_PHASE});
+
+  // set the event phase to `AT_TARGET` as in spec
+  Object.defineProperty(e, 'eventPhase', {value: Event.AT_TARGET});
+
+  // if the event bubbles through a `slot` node, then the next shadowroot host should be skipped,
+  // as the event target is in the light dom of the shadowroot host, and should not retarget
+  let throughSlot = false;
   for (let i = 0; i < path.length; i++) {
     node = path[i];
+    if (node.localName === 'slot') {
```

Comment:
I think rather than depending on traversing through `slot` here, it would be more explicit to (1) keep track of the root on which the last event was generated, (2) only generate an event if you encounter a node that has a shadowRoot that is that tracked root.